types/module/testutil/codec.go

```diff
@@ -25,12 +25,13 @@ type TestEncodingConfig struct {
 func MakeTestEncodingConfig(modules ...module.AppModule) TestEncodingConfig {
 	aminoCodec := codec.NewLegacyAmino()
 	interfaceRegistry := testutil.CodecOptions{}.NewInterfaceRegistry()
-	codec := codec.NewProtoCodec(interfaceRegistry)
+	cdc := codec.NewProtoCodec(interfaceRegistry)
+	signingCtx := cdc.InterfaceRegistry().SigningContext()
 
 	encCfg := TestEncodingConfig{
 		InterfaceRegistry: interfaceRegistry,
-		Codec:             codec,
-		TxConfig:          tx.NewTxConfig(codec, tx.DefaultSignModes),
+		Codec:             cdc,
+		TxConfig:          tx.NewTxConfig(cdc, signingCtx.AddressCodec(), signingCtx.ValidatorAddressCodec(), tx.DefaultSignModes),
```

Comment:
I have a doubt about this, if another chain use this helper, the address codec will have the wrong bech32 prefix as we provide no way to overwrite it.

Reply:
Maybe we can add an options struct to provide address codec and when empty we use those?

Reply:
I think a struct containing both prefixes will make it as CodecOptions prefixes can be overwritten before calling `NewInterfaceRegistry()`
https://github.com/cosmos/cosmos-sdk/blob/7155a1cd5278879dd7a747e39888239959bf9083/types/module/testutil/codec.go#L25-L27

Another option is to pass `CodecOptions` directly as function argument. Worst part of this is `codec/testutil` will bleed in 100+ places 😱
<img width="641" alt="Screenshot 2024-03-01 at 17 31 17" src="https://github.com/cosmos/cosmos-sdk/assets/6096878/77054b5d-bb01-4776-865b-487bbb0c1224">
What do you think?

Reply:
Mmh, that's a tricky one, breaking the api would be annoying indeed..
Given that `CodecOptions` accepts prefixes, I think we should instead use `NewInterfaceRegistryWithOptions`.
Maybe we should create a MaketestEncodingConfigWithOptions that access the two required address codecs?

Reply:
Opted to pass codecoptions as arguments. Zones can set it with its requieres prefixes.